Extensions/internet connectivity.json

```diff
@@ -0,0 +1,40 @@
+{
+  "author": "The Gem Dev",
+  "description": "",
+  "extensionNamespace": "",
+  "fullName": "Internet connectivity",
+  "name": "Internet_Connectivity",
+  "shortDescription": "an extension which checks if the device is connected to the internet. ",
+  "tags": "javascript, desktop, mobile, internet, connection,",
+  "version": "0.0.1",
+  "eventsFunctions": [
+    {
+      "description": "checks if the device is connected to the internet, when inverted it's set to not connected to the internet.",
+      "fullName": "IsOnline",
+      "functionType": "Condition",
+      "name": "isOnline",
+      "sentence": "is Online",
+      "events": [
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::Standard",
+          "conditions": [],
+          "actions": [],
+          "events": []
+        },
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "var status = runtimeScene.getVariables().get(\"connectionState\");\n\nwindow.addEventListener(\"online\", connectionStatus);\n\nfunction connectionStatus() {\n    \n    if (navigator.onLine) {\n        return\n    }\n}\n\neventsFunctionContext.returnValue = navigator.onLine;\n\n//made by TheGemDev (2020)\n\n",
```

Comment:
Don't put `made by TheGemDev (2020)`, the author is already written in the extension details.

You probably don't need `addEventListener`, remove it. This creates a leak by creating a new listener every time the condition is called, which can be catastrophic for memory usage.
Also remove the line referencing the variable, there is no need for it.

Reply:
thanks i would do that. i am super new to javascript and i did not know about the memory usage.

Reply:
Sure no worries. I recommend reading a bit about "closure" and "event listener" in JS. Basically in GDevelop we try to not use event listener at all unless we are forced too :)